### ISSUE TITLE
Single Page Generation

### ISSUE THREAD

**Issue by James-Hudson3010** [sphinx-contrib/confluencebuilder#279]
In many of my use cases, it find my documentation works better if I can keep it on a single confluence page. I have a fork at https://github.com/James-Hudson3010/confluencebuilder/tree/develop which provides this support and a test project at https://github.com/James-Hudson3010/test_confluence_builder which make use of the new feature. Nothing about how the confluencebuilder works has changed if 

`confluence_single_page = True
`

is not specified in a conf.py or is False.

Would you be interested in accepting a pull request for this code? If so, I can finish polishing it up a bit.

There is nothing particularly special about my implementation. I simply looked at how the singlehtml builder worked and took the relevant code from it.

**Comment by jdknight** [comment]
The introduction of a single-page feature (via `confluence_single_page`) would be a nice enhancement to this extension. Feel free to setup a pull request and I can help assist when needed (also see [CONTRIBUTING](https://github.com/sphinx-contrib/confluencebuilder/blob/master/CONTRIBUTING.rst)) -- I am sure community would like to have such a capability.

**Comment by James-Hudson3010** [comment]
Great. I will work on polishing it up.

One bug I still need to resolve is when one looks at

https://github.com/James-Hudson3010/test_confluence_builder/blob/develop/index.rst

there is the root title text:

```
Welcome to test_confluence_builder's documentation!
===================================================
```

This does not show up in the single page generation, although content just below the title will show up.

If anyone interested in the feature has any ideas on how to properly resolve it, I would be interested.

**Comment by jdknight** [comment]
> This does not show up in the single page generation, although content just below the title will show up.
> 
> If anyone interested in the feature has any ideas on how to properly resolve it, I would be interested.

Title entries for documents are removed by default. This is done to promote the title value which is applied to the respective Confluence document names. You maybe interested in the option [`confluence_remove_title`](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/latest/configuration.html#confluence-remove-title), which when configured to `False` will leave the title heading intact.

**Comment by James-Hudson3010** [comment]
Looks like I am ready to submit a pull request for the feature. I am having some trouble getting a complete test environment setup so all tox tests pass, but it looks like it may be good enough.

```
ERROR:   py27-sphinx16: commands failed
ERROR:   py27-sphinx17: commands failed
ERROR:   py27-sphinx18: commands failed
ERROR:   py35-sphinx16: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py35-sphinx17: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py35-sphinx18: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py36-sphinx16: commands failed
ERROR:   py36-sphinx17: commands failed
ERROR:   py36-sphinx18: commands failed
ERROR:   py37-sphinx16: commands failed
ERROR:   py37-sphinx17: commands failed
ERROR:   py37-sphinx18: commands failed
ERROR:   py38-sphinx16: commands failed
ERROR:   py38-sphinx17: commands failed
ERROR:   py38-sphinx18: commands failed
ERROR:   py35-sphinx20: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py35-sphinx21: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py35-sphinx22: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
ERROR:   py35-sphinx23: InvocationError: Failed to get version_info for python3.5: b"pyenv: python3.5: command not found\n\nThe `python3.5' command exists in these Python versions:\n  3.5.9\n\n"
  py36-sphinx20: commands succeeded
  py36-sphinx21: commands succeeded
  py36-sphinx22: commands succeeded
  py36-sphinx23: commands succeeded
  py37-sphinx20: commands succeeded
  py37-sphinx21: commands succeeded
  py37-sphinx22: commands succeeded
  py37-sphinx23: commands succeeded
  py38-sphinx20: commands succeeded
  py38-sphinx21: commands succeeded
  py38-sphinx22: commands succeeded
  py38-sphinx23: commands succeeded
  lint: commands succeeded
```

**Comment by James-Hudson3010** [comment]
fixed some issues with first pull request. second one should be a lot better...I think there may be issues around relying on some newer sphinx library features.

**Comment by jdknight** [comment]
@James-Hudson3010, thanks for introducing this enhancement. The pull request for this feature has been merged into `master` and will be available via PyPI in the stable stable release (v1.3+).

While the latest documentation will reflect the availability of the builder (for users using the bleeding edge sources), updates to reflect the newly added builder will not be added until a release is performed (so users using PyPI do not get confused why the single-page builder is not working).

Marking this issue as closed. For any new issues or comments on this feature, it is recommended to open a new issue reflecting the specific concerns.

**Comment by James-Hudson3010** [comment]
You are welcome. I am pleased you were interested enough in it to finish it off for me.
